epi_judge_python/test_framework/generic_test_handler.py

```diff
@@ -63,43 +64,45 @@ def parse_signature(self, signature):
         """
         This method is invoked for each row in a test data file (except the header).
         It deserializes the list of arguments and calls the user function with them.
 
         :param timeout_seconds: number of seconds to timeout
         :param metrics_override: metrics transformation for customizing metrics calculation
         :param test_args: serialized arguments
+        :param stdout_capturer: Captures stdout such that a test case's stdout is only printed if it fails.
         :return: list, that contains [result of comparison of expected and result, expected, result].
                  Two last entries are omitted in case of the void return type
         """
-        expected_param_count = len(
-            self._param_traits) + (0 if self.expected_is_void() else 1)
-        if len(test_args) != expected_param_count:
-            raise RuntimeError(
-                'Invalid argument count: expected {}, actual {}'.format(
-                    expected_param_count, len(test_args)))
-
-        parsed = [
-            param_trait.parse(json.loads(test_arg))
-            for param_trait, test_arg in zip(self._param_traits, test_args)
-        ]
-
-        metrics = self.calculate_metrics(parsed)
-        metrics = metrics_override(metrics, *parsed)
-
-        executor = TimedExecutor(timeout_seconds)
-        if self._has_executor_hook:
-            result = self._func(executor, *parsed)
-        else:
-            result = executor.run(lambda: self._func(*parsed))
-
-        if not self.expected_is_void():
-            expected = self._ret_value_trait.parse(json.loads(test_args[-1]))
-            self._assert_results_equal(expected, result)
-
-        return TestOutput(executor.get_timer(), metrics)
+        with redirect_stdout(stdout_capturer):
+            expected_param_count = len(
+                self._param_traits) + (0 if self.expected_is_void() else 1)
+            if len(test_args) != expected_param_count:
+                raise RuntimeError(
+                    'Invalid argument count: expected {}, actual {}'.format(
+                        expected_param_count, len(test_args)))
+
+            parsed = [
+                param_trait.parse(json.loads(test_arg))
+                for param_trait, test_arg in zip(self._param_traits, test_args)
+            ]
+
+            metrics = self.calculate_metrics(parsed)
+            metrics = metrics_override(metrics, *parsed)
+
+            executor = TimedExecutor(timeout_seconds)
+            if self._has_executor_hook:
+                result = self._func(executor, *parsed)
+            else:
+                result = executor.run(lambda: self._func(*parsed))
+
+            if not self.expected_is_void():
+                expected = self._ret_value_trait.parse(json.loads(test_args[-1]))
+                self._assert_results_equal(expected, result)
+
+            return TestOutput(executor.get_timer(), metrics)
```

Comment:
The only actual change is wrapping everything in a `with redirect_stdout(stdout_capturer):`. I think it's possible to just wrap 

https://github.com/adnanaziz/EPIJudge/blob/da8aaae8f079eaa6ebf10adbd46ae41932e36f14/epi_judge_python/test_framework/generic_test_handler.py#L95-L99

but I think the current method is clearer. Open to other opinions though!